src/diffusers/models/embeddings.py

```diff
@@ -683,3 +718,110 @@ def forward(
             objs = torch.cat([objs_text, objs_image], dim=1)
 
         return objs
+
+
+class SizeEmbedder(nn.Module):
+    """
+    Embeds scalar timesteps into vector representations.
+
+    Adapted from https://github.com/PixArt-alpha/PixArt-alpha/blob/master/diffusion/model/nets/PixArt_blocks.py.
+    """
+
+    def __init__(self, hidden_size, frequency_embedding_size=256):
+        super().__init__()
+        self.mlp = nn.Sequential(
+            nn.Linear(frequency_embedding_size, hidden_size, bias=True),
+            nn.SiLU(),
+            nn.Linear(hidden_size, hidden_size, bias=True),
+        )
+        self.frequency_embedding_size = frequency_embedding_size
+        self.outdim = hidden_size
+
+    def forward(self, size: torch.Tensor, batch_size: int):
+        if size.ndim == 1:
+            size = size[:, None]
+
+        if size.shape[0] != batch_size:
+            size = size.repeat(batch_size // size.shape[0], 1)
+            assert size.shape[0] == batch_size
+        current_batch_size, dims = size.shape[0], size.shape[1]
+        size = size.reshape(-1)
+        size_freq = get_timestep_embedding(
+            size, self.frequency_embedding_size, downscale_freq_shift=0, flip_sin_to_cos=True
+        ).to(size.dtype)
```

Comment:
```suggestion
        size_freq = get_timestep_embedding(
            size, self.frequency_embedding_size, downscale_freq_shift=0, flip_sin_to_cos=True
        ).to(size.dtype)
```
Can't we put this in the init so that we don't have to re-create timestep embeddings at every forward?